docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,172 @@
+
+**Cons we would like to overcome**
+
+Given how we have designed it, raw/whole metric name is not used in the hiveconfig - instead, a corresponding camelCase key is hardcoded to refer to this metric. We would want to move away from such design, and implement a different - less confusing strategy that can be expandable.
+
+#### Additional label support
+Hive can manage a lot of clusters, so we usually avoid labelling any metric with cluster name or namespace to keep the cardinality in check. So, in addition to the labels hive reports for a metric, some admins might want extra labels.
+For example, if admins can identify that a metric corresponds to a "managed" cluster, they can apply effective filters for their observability and can observe trends across only managed clusters, and also customize their alerts.
+
+**Pros for this design**
+
+The process of applying the labels to clusterdeployment, configuring those labels in hiveconfig to be reported with the metric, and using them as filters for alerts and grafana queries - is all in the hands of the admins/users and does not require any code change.
+It also puts the onus of cardinality on the admins.
+
+**Cons we would like to overcome**
+
+The way the code is written right now - we have a hardcoded hack for cluster-type label as mandatory if no additional labels are configured, and, most importantly, any additional labels configured will be applied to all the metrics which have optional label support, even though they might not be needed for all.
+We're essentially worsening the cardinality this way, and it makes adding this support to more metrics tricky. 
+There's also the caveat of parsing through fixed and optional labels - ensuring there's no overlap and maintaining the order for certain metrics which do not accept map of label-value while observing, are some complications that currently exist and probably will not be fixed as a part of this enhancement.
+
+
+### Goals
+Allow configuring additional labels, duration based threshold and expression matching support for all metrics that are related to cluster deployment, via HiveConfig.
+
+### Non-Goals
+Other metrics - like metrics reported by HiveOperator, or metrics that do not have the related cluster deployment available while reporting (for ex, some syncset, selectorsyncset and clusterpool metrics) will not be affected/changed.
+Adding labels only refers to the labels propagated from the Cluster Deployment, any other strings as labels cannot be configured this way.
+
+
+## Proposal
+Allow HiveConfig.Spec.metricsConfig to look like
+```
+HiveConfig:
+  spec:
+    metricsConfig:
+      clusterDeploymentRelatedMetrics:
+      - metrics:
+        - hive_foo_counter
+        - hive_foo_gauge
+        minimumDuration: 10m
```

Comment:
We may mention this elsewhere, but
- Non-duration-based metrics are allowed to be configured with the new hotness, right? And/but...
- If you specify this `minimumDuration` knob for a non-duration-based metric, we'll panic, right?

IOW, non-duration-based metrics can still be label-filtered and get additional labels; they just can't be restricted by duration.